pkg/events/derive/derive.go

```diff
@@ -1,55 +1,81 @@
 package derive
 
 import (
-	"fmt"
-
 	"github.com/aquasecurity/tracee/pkg/events"
 	"github.com/aquasecurity/tracee/types/trace"
 )
 
-// Use as static variable for testability reasons
-var getEventDefFunc = events.Definitions.Get
+// deriveFunction is a function prototype for a function that receives an event as
+// argument and may produce a new event if relevant.
+// It returns a derived or empty event, depending on successful derivation,
+// and an error if one occurred.
+type deriveFunction func(trace.Event) ([]trace.Event, []error)
```

Comment:
@NDStrahilevitz not sure if pertinent to this PR, but all derive functions return a single error `[]tracee.Event, error`, but we covert it to an slice inside `singleEventDeriveFunc` without needing it. Only `DeriveEvent` needs to return multiple errors. WDYT?

Reply:
Take a look at #2053, not sure when it will be merged but there is an example for this requirement there.